pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/gates/TNCudaGateCache.hpp

```diff
@@ -173,6 +173,18 @@ template <class PrecisionT> class TNCudaGateCache {
      */
     auto is_empty() const -> bool { return device_gates_.empty(); }
 
+    /**
+     * @brief Update an existing key with a new one.
+     *
+     * @param old_key The old key to be updated.
+     * @param new_key The new key to be updated.
+     */
+    void update_key(const std::size_t old_key, const std::size_t new_key) {
+        auto it = device_gates_.extract(old_key);
+        it.key() = new_key;
+        device_gates_.insert(std::move(it));
```

Comment:
You don't need this move anymore if `it` will be a rvalue:
```suggestion
        device_gates_.insert(it);
```